spec/mongo/crypt/binding/version_spec.rb

```diff
@@ -14,7 +14,7 @@
     end
 
     it 'is in the x.y.z-tag format' do
-      expect(version).to match(/\A(\d+.){2}(\d+)?(-[A-Za-z\+\d]+)?\z/)
+      expect(version).to match(/\A(\d+.){2}(\d+)?(-*[A-Za-z\+\d]+)?\z/)
```

Comment:
Could you explain what this change is for please?

Reply:
Latest version of libmongocrypt is `1.4.0alpha0+20220311gitd1f6bd4c9b`, and it does not match our regex. I reverted this change here because it is not related to this PR. I'll clarify why it happened and create a separate PR, if necessary.

Reply:
I see. This regexp actually suggests a certain version number shape, and the description on line 16 doesn't match the version with `+` in it. I would suggest either complexifying the regexp to  something like `/\A(\d+.){2}(\d+\w*)?([-+][A-Za-z\+\d]+)?\z/` (untested) to retain the shape suggestion, perhaps removing the last `+` allowance which I don't quite understand the purpose of to `/\A(\d+.){2}(\d+\w*)?([-+][A-Za-z\d]+)?\z/`,  or simplifying this drastically to match all kinds of present and future versions.

Reply:
Though I am also wondering whether we should be testing against unreleased versions which is what `1.4.0alpha0+20220311gitd1f6bd4c9b` looks like to me, the release should be `1.4.0alpha0` and perhaps it would be a better idea to test only against stable releases as we do with server versions (currently 1.3.1).

Reply:
Apparently, libmongocrypt versions should contain dash:
```
/**
 * @def MONGOCRYPT_VERSION
 * The version string describing libmongocrypt.
 * Has the form x.y.z-<pre>+<date>+git<sha>.
 */
 ```
The latest version has a malformed version, there is a bug for this - https://jira.mongodb.org/browse/MONGOCRYPT-412
I made our test pending until it is fixed.